### PR TITLE
Working webdriver connection using http-sync

### PR DESCRIPTION
**Pull request by jkrems**
cc @EndangeredMassa 

Picks up where https://github.com/groupon-testium/webdriver-http-sync/issues/25 left of.

Missing:

- [x] Clean up makefile to properly use npub
- [x] Test against testium
- [x] Properly support timeouts (at least somewhat properly)
- [x] Include raw stdout of child when parsing fails
- [x] Set up a travis build
- [x] Spawn phantom during test runs

### CODE REVIEW

**Comment by jkrems** [comment]
Also contains a quick (hacky) smoke test.

File: src/request/index.coffee [hunk 1]
```diff
@@ -30,55 +30,35 @@ NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
...
+      json: data
+    }
+    result = spawnSync nodeExec, [ workerFile ], {input}
+    process.stderr.write result.stderr
+
+    try
+      JSON.parse result.stdout.toString()
+    catch err
+      throw new Error """
+        Failed to communicate with webdriver: #{err.message}
```
Comment (EndangeredMassa) [inline src/request/index.coffee]: Can we include the raw `result.stdout` in this message as well?

File: test/driver.test.coffee [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+webTitle = 'A title'
+webHeading = 'A heading'
+testServer = path.join __dirname, 'test-server'
+
+describe 'Webdriver', ->
+  before 'example website', (done) ->
+    execFile testServer, [ '' + webPort ]
+    setTimeout done, 200
+
+  # before 'boot phantomjs', (done) ->
```
Comment (EndangeredMassa) [inline test/driver.test.coffee]: remove?

Comment (jkrems) [inline test/driver.test.coffee]: Actually once we have travis set up, this needs to be uncommented. Making a note for both.

**Comment by jkrems** [comment]
Aaand we're back to `http-sync`, just with one layer of wrapping less (no `request-sync` in the middle).

**Comment by jkrems** [comment]
Depends on https://github.com/dhruvbird/http-sync/pull/35

**Comment by jkrems** [comment]
@EndangeredMassa `http-sync@0.1.1` is out. Cleaned up the code & commits.

File: src/request.coffee [hunk 3]
```diff
@@ -30,55 +30,42 @@ NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ###
 
-{httpSync} = require 'request-sync'
-{extend} = require 'underscore'
-parseUrl = require('url').parse
+{parse: parseUrl} = require 'url'
```
Comment (EndangeredMassa) [inline src/request.coffee]: I find this syntax to be confusing. Do you prefer it over what was there?

File: src/request.coffee [hunk 4]
```diff
@@ -30,55 +30,42 @@ NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
...
+      if data? then new Buffer JSON.stringify(data), 'utf8'
+      else new Buffer ''
+
+    options = parseUrl url
+    options.host = options.hostname
+    options.method = method
+    options.headers =
+      'Content-Type': 'application/json'
+      'Content-Length': body.length
+      'X-Foo': 'bar'
```
Comment (EndangeredMassa) [inline src/request.coffee]: remove?

File: test/driver.test.coffee [hunk 5]
```diff
@@ -0,0 +1,60 @@
+'use strict'
```
Comment (EndangeredMassa) [inline test/driver.test.coffee]: remove?

File: test/driver.test.coffee [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+WebDriver = require '../'
+
+phantomPort = 4499
+phantomUrl = "http://127.0.0.1:#{phantomPort}"
+
+webPort = 4497
+webUrl = "http://127.0.0.1:#{webPort}/"
+webTitle = 'A title'
+webHeading = 'A heading'
```
Comment (EndangeredMassa) [inline test/driver.test.coffee]: remove?

File: test/request.test.coffee [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    @server.stderr.pipe process.stderr
+    setTimeout done, 200
+
+  after 'tear down test-server', ->
+    try @server?.kill()
+
+  it 'sends a GET request', ->
+    res = request webUrl
+    data = JSON.parse res.body
+    console.log data
```
Comment (EndangeredMassa) [inline test/request.test.coffee]: Can we add an assertion?

File: src/request.coffee (hunk 3 above)
Comment (jkrems) [inline src/request.coffee]: Honestly: I wasn't aware that the original version had it when I added it again, This code went through revisions that didn't include `parseUrl`. Restored the original version. Also restored the original timeout values. Weren't changed on purpose.

File: src/request.coffee (hunk 4 above)
Comment (jkrems) [inline src/request.coffee]: Removed.

File: test/driver.test.coffee (hunk 5 above)
Comment (jkrems) [inline test/driver.test.coffee]: Not a strict mode fan? ;) Removed.

File: test/driver.test.coffee (hunk 6 above)
Comment (jkrems) [inline test/driver.test.coffee]: Removed.

**Comment by jkrems** [comment]
* Made the tests properly test
* Removed unused code

**Comment by EndangeredMassa** [comment]
Looks good! Did you link it to testium and make sure testium tests still pass?

**Comment by jkrems** [comment]
Good point, definitely not the latest version. Will do first thing tomorrow.

**Comment by jkrems** [comment]
Good thing you asked, this change broke unicode support. Added a test for unicode input and the fix was actually to remove the re-encoding hack we had in place before.

**Comment by jkrems** [comment]
Anything blocking a merge & release of this?

**Comment by jkrems** [comment]
This also adds a first layer of tests, partially addressing #3.

**Comment by EndangeredMassa** [comment]
published in [1.1.0](https://github.com/groupon-testium/webdriver-http-sync/releases/tag/v1.1.0)

**Pull request merged**